### ISSUE TITLE
what is the intuition for applying warmup to the parameter accumulate?

### ISSUE THREAD

**Issue by vishnubanna** [ultralytics/yolov5#4362]
## ❔Question
what is the intuition for applying warmup to the parameter accumulate?
this effectively "lengthens" the warmup duration, as the number of warmup steps, number of times that optimizer.step() is called is not exactly 3 epochs number of steps. when training on many GPU's and a very large batch size, this code does nothing, as in accumulation is set and fixed at 1. Does this affect performance?

## Additional context
```
            # Warmup
            if ni <= nw:
                xi = [0, nw]  # x interp
                # model.gr = np.interp(ni, xi, [0.0, 1.0])  # iou loss ratio (obj_loss = 1.0 or iou)
                accumulate = max(1, np.interp(ni, xi, [1, nbs / total_batch_size]).round())
                for j, x in enumerate(optimizer.param_groups):
```

**Comment by glenn-jocher** [comment]
Based on emipirical results.
